### ISSUE TITLE
Voxelwise GLM: predicted signal, residuals, R-squared

### ISSUE THREAD

**Issue by Gilles86** [nistats/nistats#172]
I think when fitting a GLM to data, it is very useful to get access to the predicted signal, residuals and also R-squared of the whole model.

I implemented that here, please have a look:
https://github.com/Gilles86/nistats/tree/voxelwise_glm_stats

**Comment by bthirion** [comment]
I kind of agree but this has a cost. For instance, storing the models residuals entails keeping an additional  copy of all the data. I now have to handle GB-size images, and Nistats make it possible to run several models in a row. Storing too much stuff will blow up memories.
We might want to have an interface to handle this information, but I don't think that  it should be stored by the basic models (which essentially spits out betas and/or p-values). 
This is open to discussion !

**Comment by Gilles86** [comment]
You're right. That is why I implemented it using the "full" Regression Results class. You thus have to set minimize_memory for the FirstLevelModel to False for this to work, otherwise it will throw an exception.

My use case was that I wanted to filter some confounds with a GLM and check out which regions are most affected. Surprisingly, nistats is the only package that can do this in a few lines of Python and this is extremely useful in my opinion!

I will make some pull requests tomorrow.

**Comment by Gilles86** [comment]
See https://github.com/nistats/nistats/pull/173
